hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/entity/impl/ReactiveAbstractEntityPersister.java

```diff
@@ -729,19 +727,20 @@ else if ( !delegate().isModifiableEntity( entry ) ) {
 		}
 
 		// Now update only the tables with dirty properties (and the table with the version number)
-		return loop(
-				IntStream.range(0, span).filter( i-> tableUpdateNeeded[i] ),
-				table -> updateOrInsertReactive(
-						id,
-						fields,
-						oldFields,
-						table == 0 ? rowId : null,
-						propsToUpdate,
-						table,
-						oldVersion,
-						updateStrings[table],
-						session
-				)
+		return loop( 0, span,
+				table -> tableUpdateNeeded[table]
+						? updateOrInsertReactive(
+								id,
+								fields,
+								oldFields,
+								table == 0 ? rowId : null,
+								propsToUpdate,
+								table,
+								oldVersion,
+								updateStrings[table],
+								session
+						)
+						: voidFuture()
```

Comment:
See the thing about this change is that it means we get a longer chain, with a bunch of noop stages that the previous impl filtered out.

Reply:
Maybe... I don't know... I'm not sure how much we should worry about longer chains. This is  a CompletionStage that does nothing. Does it really have such a heavy impact?

Reply:
Well even if the only impact is a deeper stack trace, it's still bad, IMO.